pkg/apis/projectcalico/v3/nodestatus.go

```diff
@@ -0,0 +1,235 @@
+
+	// Routes reports routes known to the Calico BGP daemon on the node.
+	Routes CalicoNodeBGPRouteStatus `json:"routes,omitempty"`
+}
+
+// CalicoNodeAgentStatus defines the observed state of agent status on the node.
+type CalicoNodeAgentStatus struct {
+	// Bird4 represents the latest observed status of bird4.
+	Birdv4 CalicoNodeBirdStatus `json:"birdv4,omitempty"`
+
+	// Bird6 represents the latest observed status of bird6.
+	Birdv6 CalicoNodeBirdStatus `json:"birdv6,omitempty"`
+}
+
+// CalicoNodeBGPStatus defines the observed state of BGP status on the node.
+type CalicoNodeBGPStatus struct {
+	// The total number of IPv4 established bgp sessions.
+	V4NumEstablished int `json:"v4NumEstablished,omitempty"`
+
+	// The total number of IPv4 non-established bgp sessions.
+	V4NumNotEstablished int `json:"v4NumNotEstablished,omitempty"`
+
+	// The total number of IPv6 established bgp sessions.
+	V6NumEstablished int `json:"v6NumEstablished,omitempty"`
+
+	// The total number of IPv6 non-established bgp sessions.
+	V6NumNotEstablished int `json:"v6NumNotEstablished,omitempty"`
+
+	// V4Peers represents IPv4 BGP peers status on the node.
+	V4Peers []CalicoNodePeer `json:"v4Peers,omitempty"`
+
+	// V6Peers represents IPv6 BGP peers status on the node.
+	V6Peers []CalicoNodePeer `json:"v6Peers,omitempty"`
+}
+
+// CalicoNodeBGPRouteStatus defines the observed state of routes status on the node.
+type CalicoNodeBGPRouteStatus struct {
+	// V4 represents IPv4 routes on the node.
+	V4Routes []CalicoNodeRoute `json:"v4Routes,omitempty"`
+
+	// V6 represents IPv6 routes on the node.
+	V6Routes []CalicoNodeRoute `json:"v6Routes,omitempty"`
+}
+
+// CalicoNodeBirdStatus defines the observed state of bird.
+type CalicoNodeBirdStatus struct {
+	// Ready indicates if bird status is ready.
+	Ready bool `json:"ready,omitempty"`
+
+	// Bird version.
+	Version string `json:"version,omitempty"`
+
+	// Router ID used by bird.
+	RouterID string `json:"routerID,omitempty"`
+
+	// ServerTime holds the value of serverTime from bird.ctl output.
+	ServerTime string `json:"serverTime,omitempty"`
+
+	// LastBootTime holds the value of lastBootTime from bird.ctl output.
+	LastBootTime string `json:"lastBootTime,omitempty"`
+
+	// LastReconfigTime holds the value of lastReconfigTime from bird.ctl output.
+	LastReconfigTime string `json:"lastReconfigTime,omitempty"`
+}
+
+// CalicoNodePeer contains the status of BGP peers on the node.
+type CalicoNodePeer struct {
+	// IP address of the peer whose condition we are reporting.
+	PeerIP string `json:"peerIP,omitempty"`
+
+	// Type indicates whether this peer is configured via the node-to-node mesh,
+	// or via en explicit global or per-node BGPPeer object.
+	Type BGPPeerType `json:"type,omitempty"`
+
+	// The state is the bird bgp session state.
+	State string `json:"state,omitempty"`
+
+	// Since the state or reason last changed.
+	Since string `json:"since,omitempty"`
+
+	// Extra information from bird on the bgp session.
```

Comment:
I still don't really understand what sort of values will go here - could you give some examples?

I'd like to stay away from having fields that just blindly copy data from BIRD and expose it in the API